extractors-clipbyshape/terra_clipbyshape.py

```diff
@@ -190,13 +223,37 @@ def find_shape_image_files(self, files, triggering_file):
                             else:
                                 poly.AssignSpatialReference(ref_sys)
 
-                            # pylint: disable=line-too-long
                             imagefiles[onefile] = {'bounds' : poly}
-                            # pylint: enable=line-too-long
 
         # Return what we've found
         return (shapefile, shxfile, dbffile, imagefiles)
 
+    # pylint: disable=too-many-arguments
+    def update_dataset_extractor_metadata(self, connector, host, key, dsid, metadata,\
+                                          extractor_name):
+        """Adds or replaces existing dataset metadata for the specified extractor
+
+        Args:
+            connector(obj): the message queue connector instance
+            host(str): the URI of the host making the connection
+            key(str): used with the host API
+            dsid(str): the dataset to update
+            metadata(str): the metadata string to update the dataset with
+            extractor_name(str): the name of the extractor to associate the metadata with
+        """
+        meta = build_metadata(host, self.extractor_info, dsid, metadata, "dataset")
+
+        try:
+            md_len = len(clowder_dataset.download_metadata(connector, host, key, dsid, extractor_name))
+        except Exception as ex:     # pylint: disable=broad-except
```

Comment:
This always makes me nervous. I've seen this pattern bite people in production. Is it possible to narrow this down?

Reply:
In this case, it's done deliberately to determine if metadata is in clowder - if we get an exception for the check, we ignore it - no harm done and if there's something seriously wrong it'll be raised when we update the metadata in the next line.